### DISCUSSION TITLE
Cleaning up bad blocks

### DISCUSSION THREAD

**Discussion by braunsonm** [grafana/mimir#9205] [Help and support] [answered]
Unfortunately we suffered a data corruption event which corrupted a small number of blocks in S3. Mimir is failing the consistency check for these blocks but otherwise new data is queryable.

Is there a way to remove bad blocks from Mimir? I can see the IDs as they are output by the ruler when some rules are executed, but I'm not sure how to make sure these are removed from the index.

**Comment by pstibrany** [comment, marked as answer]
Hello. You can mark blocks for deletion using the `markblocks` tool (you can find binary under Mimir releases). Then blocks will be removed from the bucket-index by compactor, and will not be queried anymore. Compactor will eventually delete them too.
